Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -270,59 +277,68 @@ final class ConfigGenerator: ConfigGenerating {
             settings["BUNDLE_LOADER"] = "$(TEST_HOST)"
         }
 
         return settings
     }
 
-    private func deploymentTargetDerivedSettings(target: Target) -> SettingsDictionary {
-        guard let deploymentTarget = target.deploymentTarget else {
-            return [:]
-        }
-
+    private func destinationsDerivedSettings(target: Target) -> SettingsDictionary {
         var settings: SettingsDictionary = [:]
 
-        switch deploymentTarget {
-        case let .iOS(version, devices, supportsMacDesignedForIOS):
-            var deviceFamilyValues: [Int] = []
-            if devices.contains(.iphone) { deviceFamilyValues.append(1) }
-            if devices.contains(.ipad) { deviceFamilyValues.append(2) }
-
-            settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
-            settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
-            settings["SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD"] = supportsMacDesignedForIOS ? "YES" : "NO"
-
-            if devices.contains(.ipad), devices.contains(.mac) {
-                settings["SUPPORTS_MACCATALYST"] = "YES"
-                settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
-            } else {
-                // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
-                settings["SUPPORTS_MACCATALYST"] = "NO"
-            }
-        case let .macOS(version):
-            settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
-        case let .watchOS(version):
-            settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .tvOS(version):
-            settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .visionOS(version):
-            let deviceFamilyValues = [1, 2, 7]
-
-            settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
-
-            settings["XROS_DEPLOYMENT_TARGET"] = .string(version)
+        var deviceFamilyValues: [Int] = []
+        if target.destinations.contains(.iPhone) { deviceFamilyValues.append(1) }
+        if target.destinations.contains(.iPad) { deviceFamilyValues.append(2) }
+        if target.destinations.contains(.appleTv) { deviceFamilyValues.append(3) }
+        if target.destinations.contains(.appleWatch) { deviceFamilyValues.append(4) }
+        if target.destinations.contains(.appleVision) { deviceFamilyValues.append(7) }
+        settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
+
+        if target.destinations.contains(.macWithiPadDesign) {
+            settings["SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD"] = "YES"
+        } else {
+            settings["SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD"] = "NO"
```

Comment:
Running through all the fixtures and getting this cleaned up.  

One issue that is highlighting a logic change is around in `dependencyPlatformFilters`.  Previously if a `DeploymentTarget` was not set on a Target then it would not include any filters.  Now, since we use destinations, we're matching the logic as closely as we can, but targets that are iOS only without a set deployment target get configured with `platformFilter` in the Xcode project.   We could added a check for an iOS deployment target being set here to match the previous logic for this stage of the migration, but it definitely would be an odd set of logic in there.  It's probably fine since this logic will need to likely be rewritten as we address adjusting how targets are linked to each other based on destination anyways. 

Thoughts? [This method](https://github.com/tuist/tuist/pull/5132/files#diff-772e6f513a0f2bda8da0f5d60b405b48ad2fa5892ed3bf7ae3ab56b254a2f8b5R250) is the area im referring to.

Reply:
> Now, since we use destinations, we're matching the logic as closely as we can, but targets that are iOS only without a set deployment target get configured with platformFilter in the Xcode project. 

in Tuist or default Xcode?  We currently don't leverage `deploymentTarget` from the manifest (we set it in xcconfig) and don't have any `platformFitlers`  set in the generated projects 🤔 


> We could added a check for an iOS deployment target being set here to match the previous logic for this stage of the migration, but it definitely would be an odd set of logic in there. 

It's not an iOS special case to my knowledge, testing in a vanilla project in Xcode for `macOS`, platform filters aren't set either by default. 

For what I can see `platformFilters` are only to be set when something needs to explicitly diverge (this one framework / file needs a filter) otherwise no filters are set and defaults to all to match whatever the project is configured with. This seems to be the case even for multi-platform apps / frameworks.

Reply:
Ah I think I understood the comment now - indeed there will be some divergence due to the changes from `deploymentTarget` > `destinations` - I wonder why they were set in the first place given there wasn't any fine grained filtering support 🤔

Reply:
Found it 
https://github.com/tuist/tuist/pull/3152

Given this history, I think with the new APIs we can reason about this a bit better :)

Reply:
Would something like this work?

```swift

extension PBXBuildFile {
    
    /// Apply platform filters either `platformFilter` or `platformFilters` depending on count
    public func applyPlatformFilters(
        _ filters: PlatformFilters,
        hostDestinations: Destinations
    ) {
        guard !filters.isEmpty else { return }
        
        guard platformFiltersRequired(filters: filters, hostDestinations: hostDestinations) else {
            return
        }
        
        if filters.count == 1, let filter = filters.first {
            platformFilter = filter.xcodeprojValue
        } else {
            platformFilters = filters.xcodeprojValue
        }
    }
    
    private func platformFiltersRequired(filters: PlatformFilters, hostDestinations: Destinations) -> Bool {
        !hostDestinations.allSatisfy { filters.supports(destination: $0) }
    }
}

private extension PlatformFilters {
    func supports(destination: Destination) -> Bool {
        switch destination {
        case .iPhone:
            return contains(.ios)
        case .iPad:
            return contains(.ios)
        case .mac:
            return contains(.macos)
        case .macWithiPadDesign:
            return contains(.ios)
        case .macCatalyst:
            return contains(.macos)
        case .appleWatch:
            return contains(.watchos)
        case .appleTv:
            return contains(.tvos)
        case .appleVision:
            // TODO: ...
            return false
        case .appleVisionWithiPadDesign:
            return contains(.ios)
        }
    }
}
```

The idea being platform filters are applied only if there's an incompatibility between what the dependency supports and the destinations of the target consuming it?

I wonder if possibly a separate update is needed to make the various dependency types have `destinations` too instead of `platformFilters` - making it such that `platformFilters` are derived based on the dependency destinations and consumer destinations. 🤔

Reply:
> such that platformFilters are derived based on the dependency destinations and consumer destinations
I think this is the long term solution.  A given target should apply platform filters to a dependency if the dependency supports a subset of the platforms the target supports.

From my research `platformFilters` on `PBXBuildFile` control not just target depedencies but also conditional compilation, resource copying, and many other parts of the build process, so i think that the decision on what value to use will ultimately be the responsibility of other parts of the system than `applyPlatformFilters`.

 I'll update `dependencyPlatformFilters` on `TuistGraph.Target` to have the same logic as originally so this shouldnt introduce any changes.  If this were to add an `iOS` platform filter to iOS only targets though, this shouldn't be a breaking change for anyone, just a difference in the contents of the Xcode project.